index.js

```diff
@@ -239,18 +239,8 @@ class LedgerBridgeKeyring extends EventEmitter {
         return tx
       })
     }
-    // For transactions created by newer versions of @ethereumjs/tx
-    // Note: https://github.com/ethereumjs/ethereumjs-monorepo/issues/1188
-    // It is not strictly necessary to do this additional setting of the v
-    // value. We should be able to get the correct v value in serialization
-    // if the above issue is resolved. Until then this must be set before
-    // calling .serialize(). Note we are creating a temporarily mutable object
-    // forfeiting the benefit of immutability until this happens. We do still
-    // return a Transaction that is frozen if the originally provided
-    // transaction was also frozen.
-    const unfrozenTx = TransactionFactory.fromTxData(tx.toJSON(), { common: tx.common, freeze: false })
-    unfrozenTx.v = new ethUtil.BN(ethUtil.addHexPrefix(tx.common.chainId()), 'hex')
-    return this._signTransaction(address, unfrozenTx, tx.to.buf, (payload) => {
+
+    return this._signTransaction(address, tx, tx.to.buf, (payload) => {
```

Comment:
So am I pretty sure that, on line 254, we need to set the transaction type on `txData` before passing it to `fromTxData`

`toJSON()` does not return any type, but `fromTxData` needs a type or else it will treat the transaction as a legacy transaction.

@darkwing Did you say you were able to successfully send a transaction with the current state of the code?

Reply:
Yes:  
<img width="352" alt="Yes" src="https://user-images.githubusercontent.com/46655/130076348-e8886e4a-5c09-47bf-8350-caee28399d1b.png">

Reply:
I'm having trouble following this conversation :thinking: @danjm , it sounds like you're referring to the _deleted_ code with that comment. The new code doesn't call `fromTxData` or `toJSON`.

When you asked "were able to successfully send a transaction with the current state of the code?", what did you mean by "the current state of the code"? Do you mean this branch, or last published version of this package?

Reply:
@Gudahtt The current code does call `fromTxData` and `toJson` see the below lines. https://github.com/MetaMask/eth-ledger-bridge-keyring/pull/93/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346L257-L264

Reply:
So I was referring to the new line 254 https://github.com/MetaMask/eth-ledger-bridge-keyring/blob/d5df2185458ae859b44f9c31b66dc999eeadcfd1/index.js#L254